### PR TITLE
BrowsingHistory cache

### PR DESCRIPTION
**Pull request by andrewconner**

### CODE REVIEW

**Comment by andrewconner** [comment]
Test is coming.

File: shoebox/app/com/keepit/model/BrowsingHistory.scala [hunk 1]
```diff
@@ -41,8 +44,19 @@ trait BrowsingHistoryRepo extends Repo[BrowsingHistory] {
   def getByUserId(userId: Id[User])(implicit session: RSession): Option[BrowsingHistory]
 }
 
+case class BrowsingHistoryUserIdKey(userId: Id[User]) extends Key[BrowsingHistory] {
+  val namespace = "browsing_history_by_userid"
+  def toKey(): String = userId.id.toString
+}
+class BrowsingHistoryUserIdCache @Inject() (val repo: FortyTwoCachePlugin) extends FortyTwoCache[BrowsingHistoryUserIdKey, BrowsingHistory] {
+  val ttl = 24 hours
```
Comment (eishay) [inline shoebox/app/com/keepit/model/BrowsingHistory.scala:52]: does it mean that if the user didn't touch his computer for more then 24 hours (e.g. weekend), his first search will be very slow?
can we be a bit more forgiving here? say a week or so?

Comment (andrewconner) [inline shoebox/app/com/keepit/model/BrowsingHistory.scala:52]: Good point. I'll bump it up

File: shoebox/app/com/keepit/model/BrowsingHistory.scala [hunk 2]
```diff
@@ -62,12 +76,19 @@ class BrowsingHistoryRepoImpl @Inject() (val db: DataBaseComponent) extends DbRe
     def * = id.? ~ createdAt ~ updatedAt ~ state ~ userId ~ tableSize ~ filter ~ numHashFuncs ~ minHits ~ updatesCount <> (BrowsingHistory, BrowsingHistory.unapply _)
   }
 
+  override def invalidateCache(browsingHistory: BrowsingHistory) = {
+    browsingCache.set(BrowsingHistoryUserIdKey(browsingHistory.userId), browsingHistory)
```
Comment (eishay) [inline shoebox/app/com/keepit/model/BrowsingHistory.scala:80]: if FortyTwoCachePlugin.get will return the object it sets you can have a fluent api. i.e. return the set value

**Pull request merged**